pkg/generators/models/models.go

```diff
@@ -144,78 +147,114 @@ func (m *Model) Render(ctx context.Context, writer io.Writer) error {
 	return nil
 }
 
-func resolveAllOf(ref *openapi3.SchemaRef) (out *openapi3.SchemaRef) {
+func resolveAllOf(ref *openapi3.SchemaRef, passed passedSchemas) (out *openapi3.SchemaRef) {
+	if ref == nil {
+		ref = &openapi3.SchemaRef{}
+	}
+	if ref.Value == nil {
+		ref.Value = &openapi3.Schema{}
+	}
+
 	out = ref
+
+	if passed == nil {
+		passed = make(passedSchemas)
+	}
+
 	for _, subSchema := range ref.Value.AllOf {
-		out = mergeSchemas(out, subSchema)
+		if _, exists := passed[subSchema]; exists {
+			continue
+		}
+		passed[subSchema] = something{}
+		out = deepMerge(out, subSchema, passed)
 	}
+
 	return out
 }
 
-func mergeSchemas(left *openapi3.SchemaRef, right *openapi3.SchemaRef) (out *openapi3.SchemaRef) {
+func deepMerge(left *openapi3.SchemaRef, right *openapi3.SchemaRef, passed passedSchemas) (out *openapi3.SchemaRef) {
```

Comment:
Does it make sense to more explicitly document here and on resolveAllOf can causive a recusive loop with "resolveAllOf <-> deepMerge" but the infinite recursion is managed by this "passed" value?  It might not be so clear to us in the future

Reply:
Will do